components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/notifier/AbstractNotifier.java

```diff
@@ -20,24 +20,57 @@
 
 import com.google.gson.Gson;
 import org.apache.commons.codec.binary.Base64;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.apimgt.eventing.EventPublisher;
+import org.wso2.carbon.apimgt.eventing.EventPublisherEvent;
+import org.wso2.carbon.apimgt.eventing.EventPublisherException;
+import org.wso2.carbon.apimgt.eventing.EventPublisherFactory;
+import org.wso2.carbon.apimgt.eventing.EventPublisherType;
 import org.wso2.carbon.apimgt.impl.APIConstants;
+import org.wso2.carbon.apimgt.impl.internal.ServiceReferenceHolder;
 import org.wso2.carbon.apimgt.impl.notifier.events.Event;
+import org.wso2.carbon.apimgt.impl.notifier.exceptions.NotifierException;
 import org.wso2.carbon.apimgt.impl.utils.APIUtil;
 
 /**
  * Abstract class to handle common implementations related to notifiers
  */
 public abstract class AbstractNotifier implements Notifier {
 
-    protected void publishEventToEventHub(Event event) {
+    private static final Log log = LogFactory.getLog(AbstractNotifier.class);
+    private EventPublisher eventPublisher;
+
+    public AbstractNotifier() throws NotifierException {
+        if (Boolean.parseBoolean(System.getenv("FEATURE_FLAG_REPLACE_EVENT_HUB"))) {
+            try {
+                EventPublisherFactory eventPublisherFactory =
+                        ServiceReferenceHolder.getInstance().getEventPublisherFactory();
+                eventPublisher = eventPublisherFactory.getEventPublisher(EventPublisherType.NOTIFICATION);
+                eventPublisher.init();
+            } catch (EventPublisherException e) {
+                log.error("Could not initialize the event publisher for " + EventPublisherType.NOTIFICATION
+                        + ". Notification events might not be published properly.");
+                throw new NotifierException(e);
+            }
+        }
```

Comment:
Can we extract this piece of code to a place in imple.notification and reuse everywhere ?